jib-core/src/main/java/com/google/cloud/tools/jib/global/JibSystemProperties.java

```diff
@@ -83,18 +84,38 @@ public static boolean isUserAgentEnabled() {
    * @throws NumberFormatException if invalid values
    */
   public static void checkHttpTimeoutProperty() throws NumberFormatException {
-    String value = System.getProperty(HTTP_TIMEOUT);
+    checkNumericSystemProperty(HTTP_TIMEOUT, Range.atLeast(0));
+  }
+
+  /**
+   * Checks if {@code http.proxyPort} and {@code https.proxyPort} system properties are in the
+   * [0..65535] range.
+   *
+   * @throws NumberFormatException if invalid values
+   */
+  public static void checkProxyPortProperty() throws NumberFormatException {
+    checkNumericSystemProperty("http.proxyPort", Range.closed(0, 65535));
```

Comment:
Does it make any sense to make `http.proxyPort` and `https.proxyPort` constants?

Reply:
Yeah, I think `private` constants work here.

Reply:
On second thought, I think direct strings are fine. Used only in this line, and it is not that the strings will ever change. Does not seem worth a name as a field.